### PR TITLE
feat: support image block extraction

### LINKED ISSUE #306: FR/add partition option to return pdf images and pass it through the api
**Description**
This parameter is available in the `partition_pdf` but not through the API. With the new GPT-4V multimodal model, extracting images from source documents will be helpful. 

**To Reproduce**
> from unstructured.partition.pdf import partition_pdf
> partition_pdf(filename, strategy="hi_res", extract_images_in_pdf=True)

### PR DESCRIPTION
**Pull request by christinestraub**
Closes #306.

### Summary
- pass `extract_image_block_types` param through api
- pass `extract_image_block_to_payload` param to `unstructured` library
### Testing
```
$ pip uninstall unstructured
$ git clone https://github.com/Unstructured-IO/unstructured.git & cd unstructured
$ pip install -e .
$ make run-web-app
```
- PDF
```
curl -X 'POST' \
  'http://localhost:8000/general/v0/general' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'files=@sample-docs/embedded-images-tables.pdf' \
  -F 'strategy=hi_res' \
  -F 'extract_image_block_types=["Image", "Table"]' \
  | jq -C . | less -R
```
- Image
```
curl -X 'POST' \
  'http://localhost:8000/general/v0/general' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'files=@sample-docs/embedded-images-tables.jpg' \
  -F 'strategy=hi_res' \
  -F 'extract_image_block_types=["Image", "Table"]' \
  | jq -C . | less -R
```

### CODE REVIEW

**Review by awalker4 — APPROVED** [review]

**Comment by awalker4** [comment]
Oh, actually, can you add the param to openapi.json as well? That will add it to the sdks when it gets merged.

File: prepline_general/api/general.py [hunk 1]
```diff
@@ -367,6 +369,14 @@ def pipeline_api(
         int(m_max_characters[0]) if m_max_characters and m_max_characters[0].isdigit() else 1500
     )
 
+    extract_image_block_types = (
+        json.loads(m_extract_image_block_types[0])
+        if m_extract_image_block_types and len(m_extract_image_block_types)
+        else None
+    )
+
+    extract_image_block_to_payload = bool(extract_image_block_types)
```
Comment (Coniferish) [inline prepline_general/api/general.py:378]: I just want to double-check that `extract_image_block_to_payload` should always be `True` if `extract_image_block_types` is defined. In the unstructured repo I'm not seeing a boolean check like this and `extract_image_block_to_payload` is defaulting to `False` in all of the function definitions.

Comment (christinestraub) [reply prepline_general/api/general.py:378]: @Coniferish This is only required when extracting image blocks via API. When users extract image blocks using the unstructured library directly on local, they can set `extract_image_block_to_payload` to True (return images stored in metadata fields)  or False (save images directly to the local file system)

File: openapi.json [hunk 2]
```diff
@@ -213,6 +213,17 @@
                         "title": "Max Characters",
                         "description": "If chunking strategy is set, cut off new sections after reaching a length of n chars (hard max). Default: 1500",
                         "example": 1500
+                    },
```
Comment (awalker4) [inline openapi.json:216]: LGTM!

**Pull request merged**